Fix Bug 1288647: Bad Mozilla Hacking Links

danielhartnell opened the pull request:
## Description

- Adds appropriate redirect for `/hacking/` so a 404 is not returned.

## Bugzilla link

https://bugzilla.mozilla.org/show_bug.cgi?id=1288647

## Testing

## Checklist
- [ ] Requires l10n changes.
- [x] Related functional & integration tests passing.

## Additional Notes

I was not sure how to address the "Testing" and "Checklist" portions of this template. I did follow the [Managing Redirects](http://bedrock.readthedocs.io/en/latest/redirects.html#add-a-redirect) documentation and ran the following tests: `py.test -r a -m smoke tests/redirects/`. This returned the following results: "2033 passed, 37 deselected in 27.98 seconds" which looked good to me!

If there's anything more I can do, let me know.

alexgibson commented on bedrock/mozorg/redirects.py:
```diff
@@ -617,4 +617,7 @@ def decider(request, **kwargs):
     redirect(r'^projects/marketing(/.*)?$', 'https://wiki.mozilla.org/MarketingGuide'),
+
+    # bug 1288647
+    redirect(r'^hacking/?$', 'https://developer.mozilla.org/en-US/docs/Hacking_Firefox'),
```
There's not much on the redirected page, other than a simple link to `https://developer.mozilla.org/docs/Mozilla/Developer_guide/Introduction`. I'd suggest we redirect people there instead (note I left out `/en-US/` from the URL, since this means users will get directed to the most appropriate translation of the page).

alexgibson commented:
Thanks for the pull request @danielhartnell, this looks good 👍 

Aside from the one suggestion above, the only other thing we're missing here is a test for the new redirect you're adding. You can add one to the bottom of [this file](https://github.com/mozilla/bedrock/blob/master/tests/redirects/map_globalconf.py). Please also add a comment above the test that references the bug number. Thanks again!

danielhartnell commented:
Thanks for the feedback @alexgibson. I've added the test and adjusted the redirect URL. I'll be sure to keep localization in mind when making changes like this in the future.

Let me know if I can change anything else!

alexgibson commented:
Thanks @danielhartnell - this looks great, r+ on the code changes 👍 

One last thing we normally do before we merge is [squash multiple commits](http://bedrock.readthedocs.io/en/latest/contribute.html#squashing-your-commits) like this into a single commit. Is this something you are comfortable doing? If not, don't worry - just let me know here and I can do this manually before I merge.

Thanks again!

danielhartnell commented:
This will be the first time I've squashed multiple commits but it looks pretty straightforward based on the documentation you linked. This should be a great opportunity to learn something new so I'd love to give it a shot! I'll be able to work on it in a few hours. I'll @ mention you in another comment once that's complete.

danielhartnell commented:
@alexgibson,

I believe I have successfully squashed all 4 commits into a single commit. How's this pull request looking now?

alexgibson commented:
This looks good, great work @danielhartnell, thanks for contributing! r+ 👍
